content/2020-05-22-book-review-writing-an-interpreter-in-go/index.md

```diff
@@ -0,0 +1,581 @@
+---
+slug: 'book-review-writing-an-interpreter-in-go'
+date: '2020-05-22'
+title: 'Book Review: Writing an Interpreter in Go'
+description: 'My notes and thoughts on Writing an Interpreter in Go by Thorsten Ball'
+tagline: 'My notes and thoughts'
+published: true
+---
+
+![Writing an interpreter in go paperback on wood](./book_cover.jpg)
+
+When I started my new job, I was told that I would be the "Programming Languages" person. I knew I wanted to learn more not only about specific programming languages like Rust and Reason, but understand them on a higher level.
+
+I believe it was [@sugarpirate](https://twitter.com/sugarpirate_) who originally recommended both this book and the sequel "[Writing a Compiler in Go](https://compilerbook.com/)" by Thorsten Ball. I was hesitant to purchase the book because I had never written Go before and wasn't planning to. However, others insisted that you didn't need to know Go to read this book.
```

Comment:
Just a suggestions 😄 

```suggestion
I believe it was [@sugarpirate](https://twitter.com/sugarpirate_) who originally recommended both [this book](https://interpreterbook.com) and the sequel "[Writing a Compiler in Go](https://compilerbook.com/)" by Thorsten Ball. I was hesitant to purchase the book because I had never written Go before and wasn't planning to. However, others insisted that you didn't need to know Go to read this book.
```